saleor/graphql/discount/resolvers.py

```diff
@@ -23,16 +23,19 @@ def resolve_vouchers(info, channel_slug, **kwargs) -> ChannelQsContext:
 
 
 def resolve_sale(id, channel):
-    sale = models.Sale.objects.filter(id=id).first()
-    return ChannelContext(node=sale, channel_slug=channel) if sale else None
+    # TODO what about channel
+    # TODO use dataloaders
+    promotion = models.Promotion.objects.filter(old_sale_id=id).first()
+    return ChannelContext(node=promotion, channel_slug=channel) if promotion else None
```

Comment:
Should we still return `ChannelContext` ?
Do we need dataloaders here?

Reply:
It breaks over 100 tests, let do this in separate PR

Reply:
UPDATE: we cannot drop the channel context, because it used by `sale.currency` and `sale.discount_value` queries